refinery/core/models.py

```diff
@@ -1209,6 +1209,7 @@ def galaxy_cleanup(self):
     def cancel(self):
         """Mark analysis as cancelled"""
         self.canceled = True
+        self.terminate_file_import_tasks()
```

Comment:
Doesn't this already happen here: https://github.com/refinery-platform/refinery-platform/blob/45ab3a3d9ed1705f4675f58f6db6984c759d6ad6/refinery/analysis_manager/tasks.py#L298

Reply:
Yeah I believe you're right. I thought I observed different behavior yesterday